### ISSUE TITLE
Richman's question

### ISSUE THREAD

**Issue by RobinHankin** [RobinHankin/freealg#1]
There is a nice paper by Mark Haiman: "Non-commutative rational power series and algebraic generating functions", European Journal of Combinatorics (1993) 14(4):335-339 in which it says right at the end:

"The author wishes to thank David Richman, whose question "find the constant term of (x+y+x^-1 + y^-1)^p when x and y do not commute"

The freealg package can answer this.

**Comment by RobinHankin** [comment]
Oh yes:
```
 f <- function(n){constant(as.freealg("x+y+X+Y")^n)}
 sapply(c(2,4,6,8,10),f)
[1]     4    28   232  2092 19864
> 
```

This is sequence A035610 on Sloane's OEIS.

**Comment by RobinHankin** [comment]
and following 95877d6  `X^0` works, so we have:
```
 sapply(c(0,2,4,6,8,10),f)
[1]     1     4    28   232  2092 19864
>
```

**Comment by RobinHankin** [comment]
Some candidates for Sloane's OEIS:

```
> f <- function(u){sapply(0:12,function(n){constant(as.freealg(u)^n)})}
> f("x+y+X+Y")
 [1]      1      0      4      0     28      0    232      0   2092      0
[11]  19864      0 195352
> f("1+x+y+X+Y")
 [1]       1       1       5      13      53     181     713    2689   10661
[10]   41989  168785  680329 2770409
> f("x+y+XY")
 [1]    1    0    0    3    0    0   21    0    0  183    0    0 1773
> f("1+x+y+XY")
 [1]     1     1     1     4    13    31    82   253   757  2200  6601 20263
[13] 62098
```

and 
```
> f("x+y+z+X+Y+Z",0:8)

[1]     1     0     6     0    66     0   876     0 12786
```

which would be OEIS A130977.

**Comment by RobinHankin** [comment]
Fixed in 21ca023
